code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -475,9 +477,22 @@
 			adjustBruteLoss(30)
 
 /mob/living/simple_animal/adjustBruteLoss(damage)
-	health = Clamp(health - damage, 0, maxHealth)
-	if(health < 1)
-		Die()
+	if(!ignored_damage_types[BRUTE])
+		health = Clamp(health - damage, 0, maxHealth)
+		if(health < 1)
+			Die()

```

Comment:
Yeah, you could probably leave the if(!ignored_damage_types[TYPE]) checks in the adjustTYPELoss() procs, remove it from the attack_threshold_check proc, and just have attack_threshold_check use a switch statement that checks the damagetype it receives, and calls the respective adjustTYPELoss() proc.